### DISCUSSION TITLE
How to make my mod load after another?

### DISCUSSION THREAD

**Discussion by ProfLDZ** [FabricMC/fabric-api#2223] [Mod Dev Support]
A mod I use generates a resource pack, but my mod does some stuff with resource packs when the game starts. When the mod generates it's resource pack, it's always being placed on the very top of the load order, and I suspect that it's because it's actually being generated _after_ I've done my resource pack stuff. Is there a way I can make sure my mod starts loading after another mod is done?

**Comment by haykam821** [comment]
You should not change the mod load order. Consider an alternative solution to your problem such as using events; you have not provided enough context for a specific solution to be given, however.

**Reply by ProfLDZ** [reply]
I was hoping there'd be something in a config, a "loadAfter: mod" sort of thing. Since that approach may not be the best, do you have any recommendations for where to start looking to change the resource pack load order? Since a mod I use generates a resource pack after my mod's already enabled some resource packs, I can't control where the mod's resource pack loads. But I can't seem to find/understand the load order for resource packs.

**Reply by Player3324** [reply]
This is a missing feature in fabric api I think, resource and data packs are quite order sensitive. The only global (theoretically non-guaranteed) order is the lexicographical sorting by mod id in production.

**Reply by warjort** [reply]
There is no order, beyond the first/last of the ResourcePackProvider
Fabric (and other mods) break everything else.
https://github.com/FabricMC/fabric/issues/1671

And fabric's mod resource pack handling is really just the "big ball of mud" model.
https://www.curseforge.com/minecraft/mc-mods/nepacks

The order calculated goes through a number of "layers" before producing a list that gets saved in the options.txt
Ultimately, this the same list as gets displayed when you click options/resource packs on the main menu.
If you can't fix it manually in that screen, you won't be able to fix it in code or any other configuration.
